src/Form/PhpAgentSettingsForm.php

```diff
@@ -153,6 +153,14 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     $environment_variables = preg_split("(\r\n?|\n)", $values['env']);
     $cookies = preg_split("(\r\n?|\n)", $values['cookies']);
 
+    // The timeout field in the connection settings is saving an empty string if
+    // no value is provided. That causes Guzzle to report an error as it is
+    // expecting an integer value, when there is a value. A NULL value should be
+    // saved when no value is provided.
+    if (!($values['timeout'])) {
+      $values['timeout'] = NULL;
```

Comment:
**CHANGE REQUEST**

Please do get into the habit to not do `!()`, it's unnecessary. Should read just: `if (!$values['timeout'])`, it would be otherwise in the following case: `if (!($condition_a || $condition_b))`.

Reply:
I have made the change. Please review